boot/freeldr/freeldr/ntldr/arch/i386/winldr.c

```diff
@@ -487,6 +487,12 @@ WinLdrSetProcessorContext(void)
     pGdt[6].HighWord.Bytes.Flags2    = 0xC0;
     pGdt[6].HighWord.Bytes.BaseHi  = (UCHAR)((Pcr >> 24) & 0xff);
 
+    if (OperatingSystemVersion >= _WIN32_WINNT_VISTA)
+    {
+        // Vista requires the granularity bit of GDT entry 6 to be cleared
+        pGdt[6].HighWord.Bytes.Flags2 &= ~0x80;
```

Comment:
@ThFabba , @tkreuzer , could you please double-check this?

Reply:
I do not doubt that this is correct, but I would kindly ask the author to add some additional information in the comment as to why (especially what the difference is between Vista and 2k3).

Reply:
See commit c85ef31f.